src/NServiceBus.AcceptanceTests/Performance/ProcessingOptimizations/When_limiting_concurrency_via_both_api_and_config.cs

```diff
@@ -7,18 +7,19 @@ namespace NServiceBus.AcceptanceTests.Config
     using NServiceBus.AcceptanceTests.FakeTransport;
     using NServiceBus.Config;
     using NUnit.Framework;
+    using ScenarioDescriptors;
 
     public class When_limiting_concurrency_via_both_api_and_config : NServiceBusAcceptanceTest
     {
         [Test]
         public async Task Should_throw()
         {
-            var context = await Scenario.Define<Context>()
+            await Scenario.Define<Context>()
                 .WithEndpoint<ThrottledEndpoint>(b => b.CustomConfig(c => c.LimitMessageProcessingConcurrencyTo(10)))
                 .Done(c => c.EndpointsStarted)
+                .Repeat(r => r.For(Transports.AllAvailable.SingleOrDefault(t => t.Key == "FakeTransport")))
```

Comment:
Shouldn't we then find the root cause why UseTransport get's ignored?

On Tuesday, December 15, 2015, Sean Feldman <notifications@github.com>
wrote:

> In
> src/NServiceBus.AcceptanceTests/Performance/ProcessingOptimizations/When_limiting_concurrency_via_both_api_and_config.cs
> <https://github.com/Particular/NServiceBus/pull/3235#discussion_r47686115>
> :
>
> >                  .WithEndpoint<ThrottledEndpoint>(b => b.CustomConfig(c => c.LimitMessageProcessingConcurrencyTo(10)))
> >                  .Done(c => c.EndpointsStarted)
> > +                .Repeat(r => r.For(Transports.AllAvailable.SingleOrDefault(t => t.Key == "FakeTransport")))
>
> Because .UseTransport() seems to be ignored and we get
>
>  ----> System.InvalidCastException : Unable to cast object of type 'NServiceBus.AcceptanceTests.FakeTransport.FakeTransport'
> to type 'NServiceBus.AzureServiceBusTransport'```
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3235/files#r47686115>.
>


-- 
*#NSBcon 2015 • The "All About NServiceBus" conference is back *
*• Learn more <http://go.particular.net/l/27442/2015-08-21/5d4wzp> →*